### PR TITLE
KAFKA-2015: Enable ConsoleConsumer to use new consumer

### PR DESCRIPTION
**Pull request by benstopford**
This extends the original patch done by GZ to provide Console access to both the new and old consumer API's. The code follows a pattern similar to that already used in ConsoleProducer.

### CODE REVIEW

**Comment by onurkaraman** [comment]
Is partition-level consumption meant to be in scope for KAFKA-2015?

**Comment by benstopford** [comment]
@onurkaraman It's not specifically in the jira, although it wouldn't be hard to add. Is there a good use case for it?
Just thinking about it it'd be good to add the regex subscription in here when KAFKA-1893 PR gets committed.

File: core/src/main/scala/kafka/consumer/BaseConsumer.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+class NewShinyConsumer(topic: String, consumerProps: Properties) extends BaseConsumer {
+  import org.apache.kafka.clients.consumer.KafkaConsumer
+
+  val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](consumerProps)
+  consumer.subscribe(topic)
+  var recordIter = consumer.poll(0).iterator()
+
+  override def receive(): BaseConsumerRecord = {
+    while (!recordIter.hasNext)
+      recordIter = consumer.poll(0).iterator()
```
Comment (hachikuji) [inline core/src/main/scala/kafka/consumer/BaseConsumer.scala]: I wonder if we should use poll(Long.MAX_VALUE) here? We changed the meaning of poll(0) to mean return immediately even if nothing is available (before it meant block indefinitely until something is available).

**Comment by guozhangwang** [comment]
Is this relevant?

kafka.api.QuotasTest > testThrottledProducerConsumer FAILED
    junit.framework.AssertionFailedError: Should have been throttled
        at junit.framework.Assert.fail(Assert.java:47)
        at junit.framework.Assert.assertTrue(Assert.java:20)
        at kafka.api.QuotasTest.testThrottledProducerConsumer(QuotasTest.scala:136)

**Comment by ijuma** [comment]
@benstopford nitpick: the title of the PR should be of the form `KAFKA-2015; ...` (as described here https://cwiki.apache.org/confluence/display/KAFKA/Contributing+Code+Changes)

File: core/src/main/scala/kafka/consumer/BaseConsumer.scala (hunk 1 above)
Comment (benstopford) [inline core/src/main/scala/kafka/consumer/BaseConsumer.scala]: We could change it to be that way around. I'd say it's probably was 6 of one or 1/2 dozen of the other (the loop will block here or, if we specify the timeout, it'll block inside the kafka consumer)? Happy to change if you feel strongly. I actually took this straight from GZ's patch.

Comment (ijuma) [inline core/src/main/scala/kafka/consumer/BaseConsumer.scala]: There is a difference. As it is now, it will do a busy loop instead of parking the thread.

Comment (benstopford) [inline core/src/main/scala/kafka/consumer/BaseConsumer.scala]: Very good point. My bad. Changed.

File: core/src/test/scala/unit/kafka/tools/ConsoleConsumerTest.scala [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ */
+
+package unit.kafka.tools
+
+import kafka.consumer.{BaseConsumer, BaseConsumerRecord}
+import kafka.tools.{ConsoleConsumer, MessageFormatter}
+import org.easymock.EasyMock
+import org.junit.Assert._
+import org.junit.{After, Before, Test}
+import org.scalatest.junit.{JUnitSuite, JUnit3Suite}
```
Comment (guozhangwang) [inline core/src/test/scala/unit/kafka/tools/ConsoleConsumerTest.scala]: As @ijuma suggested, we do not use JUnit3Suite any more. Could you remove its usage here?

**Comment by benstopford** [comment]
@guozhangwang imports removed.

**Comment by guozhangwang** [comment]
Thanks for the patch, LGTM!

**Comment by b-goyal** [comment]
This extends the original patch done by GZ to provide Console access to both the new and old consumer API's. The code follows a pattern similar to that already used in ConsoleProducer.

**Pull request closed without merging**